spyderlib/restart_app.py

```diff
@@ -135,6 +145,29 @@ def main():
         time.sleep(0.2)  # Throttling control
 
     env = os.environ.copy()
+
+    # Reset spyder if needed
+    if reset:
+        # Build reset command
+        command_reset = '"{0}" "{1}" {2}'.format(python, spyder, args_reset)
+
+        # Try to reset
+        try:
+            p = subprocess.Popen(command_reset, shell=shell)
+            p.communicate()
+            pid_reset = p.pid
+        except Exception as error:
+            print(command)
+            traceback.print_exc()
+            time.sleep(15)

```

Comment:
If there is a problem here, the restart should be stopped instead of waiting 15s. This way the error will show up in the internal console.
Actually that means that you don't need to use `traceback.print_exc()`, just reraise the exception!